src/middlewared/middlewared/plugins/gluster_linux/utils.py

```diff
@@ -0,0 +1,36 @@
+def validate_gluster_jobs(self, verrors, job):
+    """
+    Gluster CLI operations need to run synchronously.
+    """
```

Comment:
Curious about why we choose to raise an error when another operation is already running instead of simply make it wait?

Reply:
I actually don't know how to make the current job wait for another job using middlewared facilities. I saw that there is a `core.wait_job` method that I can then run `wait_sync` method on but I found 2 issues.

1. the current job doesn't actually wait for the other job to finish
2. `core.get_jobs` has a global timeout of 60 seconds and I didn't see an easy way of changing that

Again, however, I might be missing something obvious that's already been baked into middlewared, I just couldn't find it.

Reply:
Thats the "lock" parameter of a job.

If all the gluster jobs use the same lock name, only one of these jobs can run at the same given time. They do not to be unique across them.

Reply:
Oh! Didn't even think about doing it that way. Great idea!